network/kademlia_test.go

```diff
@@ -670,3 +643,261 @@ func TestKademlia_SubscribeToNeighbourhoodDepthChange(t *testing.T) {
 		}
 	})
 }
+
+// TestCapabilitiesIndex checks that capability indices contains only the peers that have the filters' capability bits set
+func TestCapabilityIndex(t *testing.T) {
```

Comment:
Maybe this test method could split in several small testCases functions specifying what is been under test in each one. It is hard to follow all test cases as the function  advances.

Reply:
@kortatu I've split up the test. I am not totally sure how much clearer the split up makes the narrative. But at least I think the cap string mapping on the peers made the picture clearer.